R/Seurat_III.R

```
@@ -194,126 +194,260 @@ MarkStepRun <- function(seuratObj, name, saveFile = NULL) {
+  #                      OOSAP::Phenotyping_GeneList()$TCellSecondary,
+  #                      OOSAP::Phenotyping_GeneList()$TCellTranscription,
+  #                      OOSAP::Phenotyping_GeneList()$CD8Canonical,
+  #                      OOSAP::Phenotyping_GeneList()$MAIT,
+  #                      OOSAP::Phenotyping_GeneList()$CD8Subphenos1,
+  #                      OOSAP::Phenotyping_GeneList()$CD4Canonical,
+  #                      OOSAP::Phenotyping_GeneList()$CD4Subphenos1,
+  #                      OOSAP::Phenotyping_GeneList()$NKCanonical,
+  #                      OOSAP::Phenotyping_GeneList()$HighlyActivated))
+  # metadata = NULL
+  # maxCCAspaceDim = 20
+  # normalization.method = "LogNormalize"
+  # maxPCs2Weight = 20
+  
+  
+  if(!(method %in% c("simple", "cca"))) stop("method needs to be either simple or cca") else print(paste0("starting ", method, " merge"))
+  
+  
+  nameList <- NULL
+  if (is.null(metadata)){
+    nameList <- names(seuratObjs)
+  } else {
+    nameList <- names(metadata)
+  }
+  
+  if(method == "cca"){
+    seuratObj <- doMergeCCA(seuratObjs = seuratObjs, 
+                            nameList = nameList, 
+                            plotFigs = T, 
+                            maxCCAspaceDim = maxCCAspaceDim, 
+                            assay = assay, 
+                            normalization.method = normalization.method, 
+                            useAllFeatures = F, 
+                            includeCellCycleGenes = T, 
+                            cc.genes=cc.genes, g2m.genes.orig=g2m.genes.orig, 
+                            nVariableFeatures = nVariableFeatures, 
+                            spike.genes = spike.genes, maxPCs2Weight=maxPCs2Weight)
+  }
+  if(method == "simple") {
+    seuratObj <- doMergeSimple(seuratObjs = seuratObjs, 
+                                           nameList = nameList, 
+                                           projectName = "simplemerge")
+  }
```

I would make this an if/else tree  (not two if statements) and the final else should throw an error.  otherwise if the user supplies a bad value for method nothing happens.

---

I have a stop on this fx to check for bad input ...

  if(!(method %in% c("simple", "cca"))) stop("method needs to be either simple or cca") else print(paste0("starting ", method, " merge"))


so it will give an erro and stop if method is wrong..

---

Perhaps one could be the default that regardless of bad input. it does the default but still warns what happend...